### ISSUE TITLE
define type="numeric" by giving renderer, editor and validator

### ISSUE THREAD

**Issue by stla** [handsontable/handsontable#8460]
Hello,

 I [figured out](http://stackoverflow.com/questions/42025001/strange-behaviour-of-comma-as-decimal-separator-in-handsontable) that `numbro(x).format()` unformats `x` when `x` is not valid. This sounds unexpected and nonsensical to me (getting an unformatting when one requires a formatting...)

Therefore, I tried to write a custom `type="numeric"` with another formatting.

And I realized that, contrary to the claim written [in the doc here](http://docs.handsontable.com/0.30.0/tutorial-cell-function.html), it is *not* true that `type="numeric"` is equivalent to 
```
{
      renderer: Handsontable.NumericRenderer,
      editor: Handsontable.editors.TextEditor,
      validator: Handsontable.NumericValidator
}
```

Indeed, try this fiddle: https://jsfiddle.net/stla/wj5yy0zu/6/

If you modifiy (edit) cell (0,0), say you type the number `14`, then the new value is the number `14` as expected.

But if you modifiy cell (1,1) and type the number `14`, then the new value is the **string** `"14"`: 

![capture](https://cloud.githubusercontent.com/assets/4466543/22639577/f110f8fa-ec4e-11e6-8981-ed83836bada0.PNG)

I would appreciate a hint to get numbers, not strings, after editing.

**Comment by swistach** [comment]
Hi @stla, we moved documentation into a mono-repository. You can find more details in #8402.

**Comment by kirszenbaum** [comment]
@AMBudnik - the documentation issue is still present in the new docs: https://handsontable.com/docs/cell-type/#usage

The whole **Cell type** page needs a thorough rewrite, but it's out of scope of this issue.

From what I understand, to fix this issue we should make the following changes to the docs:
- Add info that `NumericEditor` is basically the same as `TextEditor`
- Add info that even if you set a cell's `renderer`, `editor` or `validator` separately, you still need to set the cell's `type`. From what I understand, this applies to every cell type (not just the `numeric` cell type).

Based on the above conclusions, I prepared the following PR: #9697

If my conclusions are wrong, please let me know here or in the PR, thanks!

**Comment by AMBudnik** [comment]
@kirszenbaum yes, the https://github.com/handsontable/handsontable/pull/9697 covers the subject entirely.

**Comment by warpech** [comment]
This issue was fixed in PR #9697, but then that fix was reverted in PR #9928.
